packages/plugin-survey/src/index.ts

```diff
@@ -165,165 +263,411 @@ class SurveyPlugin implements JsPsychPlugin<Info> {
     ranking: "ranking",
     rating: "rating",
     text: "text",
+    comment: "comment", // not listed in the jsPsych docs for simplicity, but needed here for validating question types
   };
 
-  // TO DO:
-  // - move the setup functions outside of the trial method?
-  // - add other and none options to multi-choice/select
+  // available parameters for each question type
+  private all_question_params_req = ["type", "prompt"];
+  private all_question_params_opt = ["name", "required"];
+  private all_question_params = this.all_question_params_req.concat(this.all_question_params_opt);
+  private dropdown_params = this.all_question_params.concat([
+    "options",
+    "option_reorder",
+    "add_other_option",
+    "other_option_text",
+    "dropdown_select_prompt",
+    "correct_response",
+  ]);
```

Comment:
You could also write  `private dropdown_params = [...this.all_question_params, "options",` ... here :)

Reply:
Oooh, better! I still forget about the spread operator.